packages/serverpod/lib/src/database/extensions.dart

```diff
@@ -246,3 +247,94 @@ extension ForeignKeyComparisons on ForeignKeyDefinition {
         other.referenceTableSchema == referenceTableSchema;
   }
 }
+
+/// SQL code generation methods for [Filter].
+extension FilterGenerator on Filter {
+  /// Generates a SQL WHERE clause from the filter.
+  String toQuery(TableDefinition tableDefinition) {
+    var expressions = <String>[];
+    for (var constraint in constraints) {
+      var columnDefinition = tableDefinition.findColumnNamed(constraint.column);
+      if (columnDefinition == null) {
+        throw Exception(
+          'Column "${constraint.column}" not found in table '
+          '"${tableDefinition.name}".',
+        );
+      }
+
+      var expression = constraint.toQuery(columnDefinition);
+      expressions.add('($expression)');
+    }
+
+    return expressions.join(' AND ');
+  }
+}
+
+/// SQL code generation methods for [FilterConstraint].
+extension FilterConstraintGenerator on FilterConstraint {
+  /// Generates a SQL WHERE clause from the filter constraint.
+  String toQuery(ColumnDefinition columnDefinition) {
+    assert(columnDefinition.name == column);
+
+    var columnType = columnDefinition.columnType;
+    String formattedValue;
+    if (columnType == ColumnType.integer ||
+        columnType == ColumnType.bigint ||
+        columnType == ColumnType.doublePrecision ||
+        columnType == ColumnType.boolean) {
+      formattedValue = value;
+    } else if (columnType == ColumnType.text) {
+      formattedValue = DatabasePoolManager.encoder.convert(value);
+    } else if (columnType == ColumnType.timestampWithoutTimeZone) {
+      if (type == FilterConstraintType.inThePast) {
+        formattedValue = _microsecondsToInterval(int.parse(value));
+      } else {
+        var dateTime = DateTime.tryParse(value);
+        if (dateTime != null) {
+          formattedValue = DatabasePoolManager.encoder.convert(dateTime);
+        } else {
+          formattedValue = 'NULL';
+        }
+      }
+    } else {
+      throw Exception(
+        'Unsupported column type ${columnDefinition.columnType} for column '
+        '"${columnDefinition.name}".',
+      );
+    }
+
+    if (type == FilterConstraintType.equals) {
```

Comment:
Suggestion: Use a switch statement here instead. If an enum value is not covered in a switch statement we will get a warning from the dart analyzer!